### ISSUE TITLE
Gallery block: inline padding applied on nested images (`<figure>` tag) when Gallery is full aligned

### ISSUE THREAD

**Issue by webmandesign** [WordPress/gutenberg#52721]
### Description

With Gutenberg 16.2.1 when I add a Gallery block into post/page content and make the gallery full aligned, the nested `<figure>` tags (for each individual gallery image) has horizontal padding applied.

This happens in block theme that has these `theme.json` settings (among others):
```json
{
	"$schema" : "https://schemas.wp.org/trunk/theme.json",
	"version" : 2,

	"settings" : {
		"useRootPaddingAwareAlignments" : true
	},
	"styles" : {
		"spacing" : {
			"padding"  : {
				"top"    : "0",
				"bottom" : "0",
				"right"  : "4vw",
				"left"   : "4vw"
			}
		}
	}
}
```

Everything is fine on WordPress 6.2.2 without Gutenberg plugin.

### Step-by-step reproduction instructions

1. Use appropriate theme (see description).
2. Add a Gallery block to the post content.
3. Make the Gallery full aligned.
4. Observe gaps in gallery images.

### Screenshots, screen recording, code snippet

![gallery block issue](https://github.com/WordPress/gutenberg/assets/2456735/96da0a5a-1e91-4216-b437-9f8a3850984f)

### Environment info

- WordPress 6.2.2
- Gutenberg 16.2.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by tellthemachines** [comment]
I can confirm this issue and it's not exclusive to the Gallery block but happens with the children of any full-width block with horizontal flex layout. 

A possible fix would be adding an exception for flex blocks in the root padding rule.

**Comment by webmandesign** [comment]
@annezazu could this be fixed in WP6.3 by any chance?

**Comment by tellthemachines** [comment]
@webmandesign unfortunately it won't be fixed for 6.3 as we're too close to the release date (next week) and changes to root padding code have high potential of causing further bugs, so they need to be thoroughly tested. At this point 6.4 is the likeliest candidate.

**Comment by webmandesign** [comment]
@tellthemachines Thanks for the info. I will mitigate in the theme meanwhile.
